File: Camera/Camera/SelfieTableViewController.m
//
//  SelfieTableViewConroller.m
//  Camera
//
//  Created by Joe E. on 10/19/15.
//  Copyright © 2015 Joe E. All rights reserved.
//

#import "SelfieTableViewController.h"
#import "SelfieTableViewCell.h"
#import <Parse/Parse.h>

@interface SelfieTableViewController ()

@end

@implementation SelfieTableViewController

{
    NSMutableArray *selfies;
}

- (void) logout {
    
    [PFUser logOut];
    
    UIStoryboard * userStoryboard = [UIStoryboard storyboardWithName:@"User" bundle:nil];
    UINavigationController *nc = [userStoryboard instantiateInitialViewController];
    
    [UIApplication sharedApplication].windows[0].rootViewController = nc;
    
}

- (void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"viewDidLoad");
    
    UIBarButtonItem * logoutButton = [[UIBarButtonItem alloc] initWithTitle:@"logout" style:UIBarButtonItemStylePlain target:self action:@selector(logout)];
    
//    self.navigationController.navigationItem.rightBarButtonItem = logoutButton;
    self.navigationItem.leftBarButtonItem  = logoutButton;
    
    NSLog(@"%@", self.navigationController.navigationItem);
    NSLog(@"%@", self.navigationItem);
    
    selfies = [@[] mutableCopy];
    
    PFQuery * selfieQuery = [PFQuery queryWithClassName:@"Selfie"];
    [selfieQuery findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
        
        for (PFObject * selfie in objects) {
            
            NSLog(@"%@", selfie);
            [selfies addObject:selfie];
        }
        NSLog(@"queryDidFinish");
        [self.tableView reloadData];
        
    }];
    
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    NSLog(@"%@", selfies);
    NSLog(@"current selfies count: %d", selfies.count);
    return selfies.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    SelfieTableViewCell * cell = [tableView dequeueReusableCellWithIdentifier:@"SelfieCell" forIndexPath:indexPath];
    cell.selfie = selfies[indexPath.row];
    return cell;
    
}

@end

File: Selfie/Selfie/SelfieTableViewController.swift
//
//  SelfieTableViewController.swift
//  Selfie
//
//  Created by Joe E. on 11/1/15.
//  Copyright © 2015 Joe E. All rights reserved.
//

import UIKit
import Parse

private let USER = "user"
private let LOGOUT = "logout"
private let VIEWDIDLOAD = "viewDidLoad"
private let SELFIE = "Selfie"

class SelfieTableViewController: UITableViewController {
    var selfies:NSMutableArray = NSMutableArray()
    
    func logout() {
        PFUser.logOut()
        let nc = UIStoryboard(name: USER, bundle: nil).instantiateInitialViewController()
        UIApplication.sharedApplication().windows[0].rootViewController = nc
        
        
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        NSLog(VIEWDIDLOAD)
        
        let logoutButton = UIBarButtonItem(title: LOGOUT, style: UIBarButtonItemStyle.Plain, target: self, action: "logout")
        self.navigationItem.leftBarButtonItem = logoutButton
        
        selfies = [mutableCopy()]
        
        let selfieQuery = PFQuery(className: SELFIE)
        selfieQuery.findObjectsInBackgroundWithBlock { (selfies: [PFObject]?, error:NSError?) -> Void in
            if error == nil {
                for selfie in selfies! {
                    NSLog("%@", selfie)
                    self.selfies.addObject(selfie)
                
            }
                NSLog("queryDidFinish")
                self.tableView.reloadData()

            }
            
        }

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()

    }

    // MARK: - Table view data source

    override func tableView(tableView: UITableView, numberOfRowsInSection
        section: Int) -> Int {
            NSLog("%@", selfies)
            NSLog("current selfies count: %d", selfies.count)
            return selfies.count
    }


    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> SelfieTableViewCell {
        let cell = SelfieTableViewCell()
        if let cell = tableView.dequeueReusableCellWithIdentifier("SelfieCell", forIndexPath: indexPath) as? SelfieTableViewCell {
                cell.selfie = selfies[indexPath.row] as! PFObject
            
        }
        
        return cell
        
    }

}